src/idl_gen_swift.cpp

```diff
@@ -767,8 +767,7 @@ class SwiftGenerator : public BaseGenerator {
 
     if (!IsEnum(vectortype)) code_ += const_string + "\\";
 
-    if (IsScalar(vectortype.base_type) && !IsEnum(vectortype) &&
-        !IsBool(field.value.type.base_type)) {
+    if (IsScalar(vectortype.base_type) && !IsBool(field.value.type.base_type)) {
       code_ +=
```

Comment:
This should still not allow enums since enums have their own way of returning their object:
```cpp
    if (IsEnum(vectortype)) {
      code_.SetValue("BASEVALUE", GenTypeBasic(vectortype, false));
      code_ += "return o == 0 ? {{VALUETYPE}}" + GenEnumDefaultValue(field) +
               " : {{VALUETYPE}}(rawValue: {{ACCESS}}.directRead(of: "
               "{{BASEVALUE}}.self, offset: {{ACCESS}}.vector(at: o) + "
               "index * {{SIZE}})) }";
      return;
      }
```
This will eventually depend on what we go with in my comment ofc